ctk/common/src/main/kotlin/org/codice/compliance/verification/binding/PostFormVerifier.kt

```diff
@@ -0,0 +1,312 @@
+                }
+                if (it.hasNoAttributeWithNameAndValue(METHOD, POST)) {
+                    throw SAMLComplianceException.create(
+                            SAMLBindings_3_5_4_d,
+                            message = """The form "method" is incorrect.""" +
+                                    "\n$IDP_ERROR_RESPONSE_REMINDER_MESSAGE")
+                }
+            }
+            samlResponseFormControl?.let {
+                if (it.hasNoAttributeWithNameAndValue(NAME, SAML_RESPONSE)) {
+                    throw SAMLComplianceException.create(
+                            SAMLBindings_3_5_4_b,
+                            message = "The SAMLResponse form control was incorrectly named." +
+                                    "\n$IDP_ERROR_RESPONSE_REMINDER_MESSAGE")
+                }
+                if (it.isNotHidden()) {
+                    throw SAMLComplianceException.create(
+                            SAMLBindings_3_5_4_a,
+                            message = "The SAMLResponse form control was not hidden." +
+                                    "\n$IDP_ERROR_RESPONSE_REMINDER_MESSAGE")
+                }
+            }
+            if (isRelayStateGiven) {
+                relayStateFormControl?.let {
+                    if (isRelayStateGiven && it.hasNoAttributeWithNameAndValue(NAME, RELAY_STATE)) {
+                        throw SAMLComplianceException.create(
+                                SAMLBindings_3_5_4_c,
+                                message = "The RelayState form control was incorrectly named." +
+                                        "\n$IDP_ERROR_RESPONSE_REMINDER_MESSAGE")
+                    }
+                    if (it.isNotHidden()) {
+                        throw SAMLComplianceException.create(
+                                SAMLBindings_3_5_4_c,
+                                message = "The RelayState form control was not hidden." +
+                                        "\n$IDP_ERROR_RESPONSE_REMINDER_MESSAGE")
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Verifies the relay state according to the post binding rules in the binding spec
+     * 3.5.3 RelayState
+     */
+    private fun verifyPostRelayState() {
+        if (relayState != null && relayState.toByteArray().size > MAX_RELAY_STATE_LEN)
+            throw SAMLComplianceException.createWithPropertyMessage(SAMLBindings_3_5_3_a,
+                    property = RELAY_STATE,
+                    actual = relayState)
+
+        if (isRelayStateGiven) {
+            if (relayState != EXAMPLE_RELAY_STATE) {
+                throw SAMLComplianceException.createWithPropertyMessage(SAMLBindings_3_5_3_b,
+                        property = RELAY_STATE,
+                        actual = relayState,
+                        expected = EXAMPLE_RELAY_STATE)
+            }
+        }
+    }
+
+    private fun Node.extractValue(): String? {
```

Comment:
I think we should add these to the `NodeExtensions` file. They will have to be public but that will clean up this class.